### PR TITLE
8317534: GenShen: Remove unused code

### PR DESCRIPTION
**Pull request by earthling-amzn**
Remove unused code, fix typos, remove empty statements

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JDK-8317534](https://bugs.openjdk.org/browse/JDK-8317534): GenShen: Remove unused code (**Task** - P4)

### Reviewers
 * [Y. Srinivas Ramakrishna](https://openjdk.org/census#ysr) (@ysramakrishna - Committer) ⚠️ Review applies to [6918e929](https://git.openjdk.org/shenandoah/pull/335/files/6918e92936415ab2596f274ce6ce6a4c85a26f1b)
 * [Kelvin Nilsen](https://openjdk.org/census#kdnilsen) (@kdnilsen - Committer) ⚠️ Review applies to [6918e929](https://git.openjdk.org/shenandoah/pull/335/files/6918e92936415ab2596f274ce6ce6a4c85a26f1b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah.git pull/335/head:pull/335` \
`$ git checkout pull/335`

Update a local copy of the PR: \
`$ git checkout pull/335` \
`$ git pull https://git.openjdk.org/shenandoah.git pull/335/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 335`

View PR using the GUI difftool: \
`$ git pr show -t 335`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/335.diff">https://git.openjdk.org/shenandoah/pull/335.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/shenandoah/pull/335#issuecomment-1749337055)

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahAgeCensus.hpp [hunk 1]
```diff
@@ -107,12 +107,6 @@ class ShenandoahAgeCensus: public CHeapObj<mtGC> {
   uint *_tenuring_threshold;         // An array of the last N tenuring threshold values we
                                      // computed.
 
-  // A private work method invoked by the public compute_tenuring_threshold() method.
-  // This uses the data in the ShenandoahAgeCensus object's _global_age_table and the
-  // current _epoch to compute a new tenuring threshold, which will be remembered
-  // until the next invocation of compute_tenuring_threshold.
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahAgeCensus.hpp:113]: Could you please move this documentation to `compute_tenuring_threshold()` further below. Thanks!

File: src/hotspot/share/gc/shenandoah/shenandoahGeneration.hpp [hunk 2]
```diff
@@ -135,10 +135,6 @@ class ShenandoahGeneration : public CHeapObj<mtGC>, public ShenandoahSpaceInfo {
   void increase_capacity(size_t increment);
   void decrease_capacity(size_t decrement);
 
-  void set_soft_max_capacity(size_t soft_max_capacity) {
-    _soft_max_capacity = soft_max_capacity;
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahGeneration.hpp:139]: Not only the accessor, but it seems like the `_soft_max_capacity` field in `ShenandoahGeneration` is vestigial and can be deleted?

A cursory look seemed to indicate that a field with a similar name (similar accessor, but confusingly different field name) is used for the purpose for that purpose for the whole heap, and the generation's field is obsolete/vestigial. I may have missed some uses in my inspection, but you can check if deleting the field is ok (it's only used in logging where it seems not to have much value?).

File: src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp [hunk 3]
```diff
@@ -1927,37 +1920,6 @@ void ShenandoahHeap::gclabs_retire(bool resize) {
...
-  for (JavaThreadIteratorWithHandle jtiwh; JavaThread *t = jtiwh.next(); ) {
-    cl.do_thread(t);
-    ThreadLocalAllocBuffer& tlab = t->tlab();
-    if (tlab.end() != nullptr) {
-      ShenandoahHeapRegion* r = heap_region_containing(tlab.start());
-      r->set_young_lab_flag();
-    }
-  }
-}
-
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp:1960]: Nice!

File: src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp [hunk 4]
```diff
@@ -1015,13 +1009,4 @@ inline bool ShenandoahRegionChunkIterator::next(struct ShenandoahRegionChunk *as
...
 
-template<class T>
-inline void ShenandoahVerifyNoYoungRefsClosure::work(T* p) {
-  T o = RawAccess<>::oop_load(p);
-  if (!CompressedOops::is_null(o)) {
-    oop obj = CompressedOops::decode_not_null(o);
-    assert(!_heap->is_in_young(obj), "Found a young ref");
-  }
-}
-
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp:1026]: Very curious. I could have sworn I had used this in some assertions/verification code in card scanning or card verification, but may be that code went away. Does seem to be unused now, so best to get rid of it.

File: src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp [hunk 5]
```diff
@@ -855,13 +826,10 @@ class ShenandoahHeap : public CollectedHeap {
 public:
   inline RememberedScanner* card_scan() { return _card_scan; }
   void clear_cards_for(ShenandoahHeapRegion* region);
-  void dirty_cards(HeapWord* start, HeapWord* end);
-  void clear_cards(HeapWord* start, HeapWord* end);
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp:859]: These aren't used, but in looking through the code for marking and clearing cards in `ShenandoahDirectCardMarkRememberedSet::mark_range_as_...` I realize we might want to see if we want to use vector instructions or use `memset`, when the range is sufficiently large (future work).

**Review by ysramakrishna — APPROVED** [review]
LGTM; thanks for the clean-up.

A few somewhat minor comments.

**Review by kdnilsen — APPROVED** [review]

File: src/hotspot/share/gc/shenandoah/shenandoahGeneration.hpp (hunk 2 above)
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahGeneration.hpp:139]: Hmm, this field is used by the heuristics. This looks like a regression to me. The soft max heap size can be given as a command line argument, but it is also supposed to be a 'managed' option that can be changed with jmx. I will open a separate ticket for this.

File: src/hotspot/share/gc/shenandoah/shenandoahAgeCensus.hpp (hunk 1 above)
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahAgeCensus.hpp:113]: Done.

File: src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp (hunk 5 above)
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp:859]: That's a good idea!

**Pull request closed without merging**